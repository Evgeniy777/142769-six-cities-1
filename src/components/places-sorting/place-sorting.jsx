import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {getSort, getPlacesFilter} from "../../reducer/app/selectors";
import {getFilteredOffers} from "../../reducer/data/selectors";
import {ActionCreator} from '../../reducer/app/app';

const PlaceSorting = (props) => {
  const {sort, placesFilter, onItemClick, onToggleMenu, onSorting} = props;
  const list = Object.values(placesFilter);

  const filterTitles = {
    [`POPULAR`]: `Popular`,
    [`PRICE_ASC`]: `Price: low to high`,
    [`PRICE_DESC`]: `Price: high to low`,
    [`TOP_RATED_FIRST`]: `Top rated first`
  };

  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span className="places__sorting-type" tabIndex="0" onClick={onToggleMenu}>
        {filterTitles[sort]}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select"></use>
        </svg>
      </span>
      <ul className="places__options places__options--custom">
        {list.map((it, i) => {
          const item = filterTitles[it];
          return (
            <li
              className={`places__option ${it === sort ? `places__option--active` : ``}`}
              tabIndex={i}
              key={`${item}-${i}`}
              onClick={() => {
                onItemClick(it);
                onSorting(it);
              }
              }>
              {item}
            </li>
          );
        })}
      </ul>
    </form>
  );
};

PlaceSorting.propTypes = {
  sort: PropTypes.string.isRequired,
  placesFilter: PropTypes.object.isRequired,
  onSorting: PropTypes.func,
  onItemClick: PropTypes.func,
  onToggleMenu: PropTypes.func
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    sort: getSort(state),
    placesFilter: getPlacesFilter(state),
    filteredOffers: getFilteredOffers(state)
  });
};
const mapDispatchToProps = (dispatch) => ({
  onSorting: (type) => {
    dispatch(ActionCreator.changeSort(type));
  },
  onToggleMenu: () => {
    document.querySelector(`.places__options`).classList.toggle(`places__options--opened`);
  }
});

export {PlaceSorting};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(PlaceSorting);
