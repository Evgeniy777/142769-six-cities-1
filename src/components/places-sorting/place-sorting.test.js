import React from 'react';
import renderer from 'react-test-renderer';
import {PlaceSorting} from '../places-sorting/place-sorting.jsx';

describe(`PlaceSorting`, () => {
  it(`PlaceSorting correctly renders`, () => {
    const placeSorting = [
      `Popular`,
      `Price: low to high`,
      `Price: high to low`,
      `Top rated first`
    ];
    const tree = renderer
      .create(<PlaceSorting
        placeSorting={placeSorting}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
