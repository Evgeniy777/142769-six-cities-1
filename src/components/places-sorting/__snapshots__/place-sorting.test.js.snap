// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceSorting PlaceSorting correctly renders 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    tabIndex="0"
  >
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  />
</form>
`;
